### ISSUE TITLE
Retrieve Authority Information Access via x509_certificate_info

### ISSUE THREAD

**Issue by benaryorg** [ansible-collections/community.crypto#529]
##### SUMMARY

Improve *community.crypto.x509_certificate_info* to also yield the contents of *Authority Information Access* (OID 1.3.6.1.5.5.7.1.1).

Alternatively:

- allow custom OIDs to be ASN.1 parsed by the module or
- have a filter for parsing the current output (b64'd ASN.1) to usable values within Ansible (or provide documentation on how to do so if already possible)

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

`x509_certificate_info`

##### ADDITIONAL INFORMATION

AIA is present in most if not all certificates, and it allows one to piece together a certificate chain based on a single certificate.
This is incredibly useful when only one certificate or an invalid chain is present, since it allows one to recreate a valid chain in most cases.

```yaml
- name: Get certificate information
  community.crypto.x509_certificate_info:
    path: /some/certificate.pem
  register: certinfo

- name: Get upstream certificate
  ansible.builtin.debug:
    # currently the next best thing is to read the appropriate OID, which yields base64'd ASN.1 which is hard to parse in Ansible directly
    # one *could* pass the result off to another task to fetch the certificate then
    # this prints "MEcwIQYIKwYBBQUHMAGGFWh0dHA6Ly9yMy5vLmxlbmNyLm9yZzAiBggrBgEFBQcwAoYWaHR0cDovL3IzLmkubGVuY3Iub3JnLw==" for a Let's Encrypt cert
    msg: |-
      {{ certinfo.extensions_by_oid["1.3.6.1.5.5.7.1.1"] }}

- name: Get upstream certificate (decoded)
  ansible.builtin.debug:
    # currently even base64 decoding the certificate causes issues in Ansible since either Ansible or Jinja don't seem to work well with binary values
    # this should in theory yield "lencr" from the "http://r3.i.lencr.org/" link contained, but it does yield an empty string
    msg: |-
      {{ certinfo.extensions_by_oid["1.3.6.1.5.5.7.1.1"] | b64decode | regex_search("[a-z]{5}", multiline=True) }}
```

Basically I'm looking for any means (short of writing a custom plugin/filter/module) to retrieve the `http://r3.i.lencr.org/` string from the cert.
Note that this is a different string than the `http://r3.o.lencr.org` which is the OCSP responder which is already being extracted it seems (as `certinfo.ocsp_uri`).

**Comment by felixfontein** [comment]
> since either Ansible or Jinja don't seem to work well with binary values

I think Jinja works fine, but Ansible doesn't support binary values well.

**Comment by Tyrael** [comment]
sorry for necroing this issue, but is it possible that there is still no support/helper method for decoding the base64 asn.1 custom oids from a cert?
